src/app/course/course-question-snippet/course-question-snippet.component.html

```diff
@@ -44,11 +44,75 @@ <h2 *ngIf="eventId" class="mb-3 mt-5">{{ event.name }} Questions</h2>
                                 <a class="btn btn-warning mr-3"
                                    [routerLink]="['/problem', uqj.question.id,'edit']">Edit</a>
                             </ng-container>
-
+                            <ng-container *ngIf="uqj?.report.id">
+                                <button type="button" mat-icon-button
+                                        (click)="createModal(reportModal, uqj)">
+                                    <!--(click)="reportCheck(uqj)">-->
+                                    <mat-icon>flag</mat-icon>
+                                </button>
+                            </ng-container>
+                            <ng-container *ngIf="!uqj?.report.id">
+                                <button type="button" mat-icon-button
+                                        (click)="createModal(reportModal, uqj)">
+                                    <mat-icon>outlined_flag</mat-icon>
+                                </button>
+                            </ng-container>
                         </td>
                     </tr>
                 </ng-container>
             </ng-container>
+
+
+            <ng-template #reportModal let-modal>
+                <div class="modal-header">
+                    <h4 class="modal-title" id="modal-basic-title">Report Question</h4>
+                    <button type="button" class="close" aria-label="Close" (click)="modal.dismiss()">
+                        <span aria-hidden="true">×</span>
+                    </button>
+                </div>
+                <form [formGroup]="formGroup" (ngSubmit)="reportQuestion()">
+                    <div class="modal-body">
+                        <div class="form-group">
+                            <label for="description">Report Reason:</label>
+                            <select class="form-control" name="description" formControlName="description"
+                                    id="description">
+                                <option value="TYPO_TEXT">There is a typo in the question instructions</option>
```

Comment:
Use enum values for each option

Reply:
We made the report an enum and updated these values!